database.hh

```diff
@@ -595,7 +595,7 @@ private:
     // Mutations returned by the reader will all have given schema.
     flat_mutation_reader make_sstable_reader(schema_ptr schema,
                                         reader_permit permit,
-                                        lw_shared_ptr<sstables::sstable_set> sstables,
+                                        lw_shared_ptr<sstables::sstable_set_snapshot> sstables,
```

Comment:
Yes. Will cause less churn. Also in people's mind.

Reply:
I had an impression that either way, we are creating a single new class and changing a portion of sstable_sets to it. If it's preferable to change the versioned one, I'll do just that

Reply:
The problem with your approach (if I understand it correctly) is that sstable_set changes meaning. Before, it was a set of sstables. Afterwards, it's the history of sstable_sets, from which you can derive snapshots. sstable_set_snapshot takes the meaning of the old sstable_set. That confuses everyone, because any time they see sstable_set they have to attach it to either the old or the new meaning.

Reply:
It is true that the sstable_set changed meaning. But it wouldn't be exactly true to say that the sstable_set_snapshot is a set of sstables either. After all, it doesn't contain any sstables on its own, it's just an interface for asking the versioned sstable_set for the sstables. You can't erase or insert any sstables to it too, so it would be closer to a const set.

As I said though, the approach with a versioned_sstable_set is good, so I'll change to it, especially because it may help reviewers.